[FSDP][Easy] Rename `_prefixed_param_names` -> `_fqns` for consistency

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#86653 [FSDP][Easy] Rename `_prefixed_param_names` -> `_fqns` for consistency**
* #86528 [FSDP] Remove `utils.py` (moved to `_utils.py`)
* #85738 [FSDP] Add initial `summon_full_params(with_grads=True)`
* #84911 [FSDP] Add `use_orig_params`

This renames `_prefixed_param_names` to `_fqns` to help converge on the terminology.

Commit:
[FSDP][Easy] Rename `_prefixed_param_names` -> `_fqns` for consistency

[ghstack-poisoned]

rohan-varma commented on torch/distributed/fsdp/flat_param.py:
```diff
@@ -156,14 +156,14 @@ class FlatParameter(nn.Parameter):
-            construct this flattened parameter via :class:`FlatParamHandle`;
-            the prefixed names are guaranteed to be unique within the subtree
-            rooted in that module.
+        _fqns (Tuple[str, ...]): Each parameter's name prefixed with the parent
```
Each original parameter / unflattened parameter to clarify?

rohan-varma reviewed (approved):
LGTM, thanks!

Commit:
Update on "[FSDP][Easy] Rename `_prefixed_param_names` -> `_fqns` for consistency"


This renames `_prefixed_param_names` to `_fqns` to help converge on the terminology.


[ghstack-poisoned]

awgu commented:
@pytorchbot merge

awgu commented:
@pytorchbot merge
